tests/CombinedReceive.py

```diff
@@ -0,0 +1,123 @@
+import time
+import lsst.ctrl.events as events
+from lsst.daf.base import PropertySet
+
+#
+# Send an event
+#
+class CombinedReceiveEventTestCase(unittest.TestCase):
+
+    def sendEvent(self, broker, topic):
+        trans = events.EventTransmitter(broker, topic)
+        
+        root = PropertySet()
+        root.set("TOPIC",topic)
+        root.set("myname","myname")
+        root.set("STATUS", "my special status")
+        root.set("RUNID","srptestrun")
+        
+        event = events.Event("srptestrun", root)
+    
+        # wait a short time so we can see the difference between the time 
+        # the event is created and the time it is published
+        time.sleep(2)
+    
+        # ok...now publish it
+        trans.publishEvent(event)
+
+    def testCombinedReceive(self):
+        # Test sending on two topics, and receiving bother on a third.
+        #
+        broker = "lsst8.ncsa.illinois.edu"
+        host = platform.node()
+        pid = os.getpid()
+        host_pid = "%s_%d" % (host,pid)
+    
+        topicA = "test_events_9_%s.A" % host_pid
+        topicB = "test_events_9_%s.B" % host_pid
+        topicC = "test_events_9_%s.*" % host_pid
+    
+        recvA = events.EventReceiver(broker, topicA)
+        recvB = events.EventReceiver(broker, topicB)
+        recvC = events.EventReceiver(broker, topicC)
+    
+        # send a test event on topicA and topicB
+        self.sendEvent(broker, topicA)
+        self.sendEvent(broker, topicB)
+
+        # receive event on topicA, and check to see it's the
+        # right one
+        val = recvA.receiveEvent()
+        self.assertNotEqual(val, None)
+        ps = val.getPropertySet()
+        name = ps.get("TOPIC")
+        self.assertEqual(name, topicA)
+
+        # receiving no more messages on topicA
+        val = recvA.receiveEvent(1)
+        self.assertEqual(val, None)
+
+        # receive event on topicB, and check to see it's the
+        # right one
+        val = recvB.receiveEvent()
+        self.assertNotEqual(val, None)
+        ps = val.getPropertySet()
+        name = ps.get("TOPIC")
+        self.assertEqual(name, topicB)
+
+        # receiving no more messages on topicB
+        val = recvB.receiveEvent(1)
+        self.assertEqual(val, None)
+    
+        # receive event on topicC, and check to see it's
+        # the one sent to topicA
+        val = recvC.receiveEvent()
+        self.assertNotEqual(val, None)
+        ps = val.getPropertySet()
+        name = ps.get("TOPIC")
+        self.assertEqual(name, topicA)
+    
+        # receive event on topicC, and check to see it's
+        # the one sent to topicB
+        val = recvC.receiveEvent()
+        self.assertNotEqual(val, None)
+        ps = val.getPropertySet()
+        name = ps.get("TOPIC")
+        self.assertEqual(name, topicB)
+
+        # receiving no more messages on topicC
+        val = recvC.receiveEvent(1)
+        self.assertEqual(val, None)
+    
+
+if __name__ == "__main__":
```

Comment:
Tests in afw, etc, have something like:

    import lsst.utils.tests as utilsTests
    
    […]
    def suite():
        """Returns a suite containing all the test cases in this module."""
    
        utilsTests.init()
        suites = []
        suites += unittest.makeSuite(ImageTestCase)
        suites += unittest.makeSuite(DecoratedImageTestCase)
        suites += unittest.makeSuite(utilsTests.MemoryTestCase)
        return unittest.TestSuite(suites)
    
    def run(shouldExit=False):
        """Run the tests"""
        utilsTests.run(suite(), shouldExit)
    
    if __name__ == "__main__":
        run(True)


The `lsst.utils.tests.MemoryTestCase` checks for memory leaks of objects that are subclassed from `Citizen`.